### ISSUE TITLE
Travis reports job passing although unit tests failed

### ISSUE THREAD

**Issue by beat-buesser** [IBM/adversarial-robustness-toolbox#95]
It seems that Travis reports build jobs as passing although unit tests report errors, e.g.: 
https://travis-ci.org/IBM/adversarial-robustness-toolbox/jobs/547884245

**Comment by ririnicolae** [comment]
@beat-buesser I had also noticed this recently. This happens because the unit tests are launched through multiple commands and Travis only catches the code from the last command that ran. I have a potential fix that chains the commands in order to transmit all the exit codes to Travis. Are you working to fix this or should I?

**Comment by beat-buesser** [comment]
@ririnicolae I have just pushed a possible solution. Sorry, I didn't see your response before pushing. Let me know if you have a more elegant solution.

**Comment by beat-buesser** [comment]
The unit tests seem to again get analysed correctly by Travis.

Following this, I have discovered two unit tests (below) that are faiingl since #68 which has been merged a few days after this bug has been introduced. It seems pickling of pytorch and tensorflow classifiers is failing for python2. Since we don't plan to close #83 (sunset python2 support) before the next release I'll take a look at fixing them and talk with @minhitbk about possible quick solutions.

```
======================================================================
FAIL: test_pickle (test_pytorch.TestPyTorchClassifier)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/IBM/adversarial-robustness-toolbox/tests/classifiers/test_pytorch.py", line 243, in test_pickle
    self.assertTrue(self.module_classifier.__dict__.keys() == loaded.__dict__.keys())
AssertionError: False is not true
======================================================================
FAIL: test_pickle (test_tensorflow.TestTFClassifier)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/IBM/adversarial-robustness-toolbox/tests/classifiers/test_tensorflow.py", line 190, in test_pickle
    self.assertTrue(self.classifier.__dict__.keys() == loaded.__dict__.keys())
AssertionError: False is not true
```

**Comment by beat-buesser** [comment]
Another unit test related to this issue fixed in 0685ebdabe3203c36cd8db1eab6f52b6e497304a as part of #93 :

```
======================================================================
ERROR: test_mnist (test_iterative_method.TestIterativeAttack)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/IBM/adversarial-robustness-toolbox/tests/attacks/test_iterative_method.py", line 76, in test_mnist
    self._test_backend_mnist(classifier)
  File "/home/travis/build/IBM/adversarial-robustness-toolbox/tests/attacks/test_iterative_method.py", line 111, in _test_backend_mnist
    attack = BasicIterativeMethod(classifier, num_random_init=3)
TypeError: __init__() got an unexpected keyword argument 'num_random_init'
````
